lib/utils/sequelize-method.ts

```diff
@@ -0,0 +1,129 @@
+import isObject from 'lodash/isObject';
+import type { AttributeType, LogicType } from '../..';
+
+/**
+ * Utility functions for representing SQL functions, and columns that should be escaped.
+ * Please do not use these functions directly, use Sequelize.fn and Sequelize.col instead.
+ *
+ * @private
+ */
+export class SequelizeMethod {}
+
+/**
+ * Do not use me directly. Use {@link Sequelize.fn}
+ */
+export class Fn extends SequelizeMethod {
+  private readonly fn: string;
+  private readonly args: unknown[];
+
+  constructor(fn: string, args: unknown[]) {
+    super();
+    this.fn = fn;
+    this.args = args;
+  }
+
+  clone(): Fn {
+    return new Fn(this.fn, this.args);
+  }
+}
+
+/**
+ * Do not use me directly. Use {@link Sequelize.col}
+ */
+export class Col extends SequelizeMethod {
+  private readonly col: string[] | string;
+
+  constructor(col: string[] | string, ...args: string[]) {
+    super();
+    // TODO(ephys): this does not look right. First parameter is ignored if a second parameter is provided.
```

Comment:
Is this something that you want to address in this PR?

Reply:
I'd rather do a follow-up PR, to make sure I'm not breaking something deliberate